disnake/member.py

```diff
@@ -722,17 +722,39 @@ def current_timeout(self) -> Optional[datetime.datetime]:
 
         return self._communication_disabled_until
 
+    @overload
+    async def ban(
+        self,
+        *,
+        clean_history_duration: Union[int, datetime.timedelta] = 86400,
+        reason: Optional[str] = None,
+    ) -> None:
+        ...
+
+    @overload
     async def ban(
         self,
         *,
-        delete_message_days: Literal[0, 1, 2, 3, 4, 5, 6, 7] = 1,
+        delete_message_days: Literal[0, 1, 2, 3, 4, 5, 6, 7],
+        reason: Optional[str] = None,
+    ) -> None:
+        ...
+
+    async def ban(
+        self,
+        *,
+        clean_history_duration: Union[int, datetime.timedelta] = 86400,
+        delete_message_days: Literal[0, 1, 2, 3, 4, 5, 6, 7] = MISSING,
         reason: Optional[str] = None,
     ) -> None:
         """|coro|
 
         Bans this member. Equivalent to :meth:`Guild.ban`.
         """
-        await self.guild.ban(self, reason=reason, delete_message_days=delete_message_days)
+        if delete_message_days is not MISSING:
+            await self.guild.ban(self, reason=reason, delete_message_days=delete_message_days)
+        else:
+            await self.guild.ban(self, reason=reason, clean_history_duration=clean_history_duration)
```

Comment:
in order to get the proper error handling with guild.ban and TypeError, we need to pass through all of the parameters exactly as they are provided. A #type: ignore comment should be used in this case.

Reply:
https://github.com/DisnakeDev/disnake/pull/659/commits/ea7e8a0db4583294cbe48728384ea326340454a7